Assets/HexMap.cs

```diff
@@ -226,96 +201,54 @@ public Hex GetAcceptableStartPosition()
-                    h.SetTerrain(Terrain.MOUNTAIN);
-                }
-                else if (h.Elevation >= HeightHill)
-                {
-                    mf.mesh = MeshHill;
-                    walkableHexes.Add(h);
-                    h.SetTerrain(Terrain.HILLS);
-                }
-                else if (h.Elevation >= HeightFlat)
-                {
-                    mf.mesh = MeshFlat;
-                    walkableHexes.Add(h);
-                }
-                else
-                {
-                    mr.material = MatOcean;
-                }
-
-                if (h.Elevation >= HeightMountain)
-                    mr.material = MatMountains;
-                else if (h.Elevation < HeightFlat)
-                {
-                    mf.mesh = MeshWater;
-                    mr.material = MatOcean;
-                    h.SetTerrain(Terrain.WATER);
-                }
-                else if (h.Moisture >= MoistureRainforest)
-                {
-                    mr.material = MatGrasslands;
-
-                }
-                else if (h.Moisture >= MoistureForest)
-                {
-                    mr.material = MatGrasslands;
-                    
-                }
-                else if (h.Moisture >= MoistureGrassland)
-                {
-                    mr.material = MatGrasslands;
-                    
-                }
-                else
-                {
-                    mr.material = MatDesert;
-                }
-                if(mf.mesh != MeshHill)
-                {
-                    if (mr.material == MatGrasslands)
-                        h.SetTerrain(Terrain.GRASS);
-                    else if (mr.material == MatDesert)
-                        h.SetTerrain(Terrain.DESERT);
-                }
+                results.Add(GetHexAt(centerHex.Q + dx, centerHex.R + dy));
             }
         }
+        return results.ToArray();
     }
 
-    public Hex[] GetHexesWithRadiusOf(Hex centerHex, int radius)
+    protected void SetTerrainForHex(Hex hex)
     {
-        List<Hex> results = new List<Hex>();
-        for (int dx = -radius; dx < radius; dx++)
+        if (hex.Elevation >= HeightMountain)
+            hex.SetTerrain(Terrain.MOUNTAIN);
+        else if (hex.Elevation >= HeightHill)
         {
-            for (int dy = Mathf.Max(-radius, -dx - radius); dy < Mathf.Min(radius, -dx + radius); dy++)
+            walkableHexes.Add(hex);
+            if (hex.Moisture >= MoisturePlains)
+                hex.SetTerrain(Terrain.GRASSY_HILLS);
+            else
+                hex.SetTerrain(Terrain.ARID_HILLS);
+        }
+        else if (hex.Elevation >= HeightFlat)
+        {
+            walkableHexes.Add(hex);
+            if (hex.Moisture >= MoisturePlains)
+                hex.SetTerrain(Terrain.GRASS);
+            else
+                hex.SetTerrain(Terrain.DESERT);
+        }
+        else
+            hex.SetTerrain(Terrain.WATER);
+    }
+
```

Comment:
Is this a WIP? The logic needs to go in its own manager / object or the like. Keep track of this kind of flow manually is painful.